src/main/java/com/eternalcode/parcellockers/parcel/ParcelLockerRepositoryJdbcImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.eternalcode.parcellockers.parcel;
+
+import com.eternalcode.parcellockers.database.JdbcConnectionProvider;
+import com.eternalcode.parcellockers.util.LocationUtil;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+public class ParcelLockerRepositoryJdbcImpl implements ParcelLockerRepository {
```

Comment:
Generalnie jak nie chcesz opakowywać tych metod w future to i tak przydało by się wykonanie wszystkich poleceń do bazy danych async. Np. przy użyciu 
```ExecutorService executor = Executors.newSingleThreadExecutor(); (nie jestem pewny co do pisowni)

i później executor.submit(() -> <tutaj wrzucasz statementa który leci do bazy>);
 ```

Komentarz dodany po tych dwóch na dole.

Reply:
@Jakubk15 tu wleciała lekka poprawka bo ucieło górą część wcześniej